### ISSUE TITLE
Avoiding inputs of incorrect lengths

### ISSUE THREAD

**Issue by umihoshijima** [jpgattuso/seacarb-git#53]
In going through the function code, I became curious about the way that seacarb handles input variables of varying lengths. Currently, all functions look for the input variable with the maximum length, then force all shorter variables to that length by repeating the first value of that variable. This does deal with the majority of situations, where input values _should_ be either the same length, or 1. However, if for some reason the user passes two inputs of different lengths (e.g. 4 and 5), the current way of dealing with input vector lengths would not throw an error, nor would it return 4 values. It would instead use the first value in the input of length 4 for all five calculations. 

The matlab CO2SYS does things a similar way, but has another measure to stop this from occurring; that is, they make sure that there are no more than two unique vector lengths being inputted. The idea with this is to allow for constant inputs (e.g. all calculations with constant salinity), with all other values being equal. However, it would not throw an error for the first case presented (inputs of length 4 and 5). 

The solution here is possibly to implement a line of code such that inputs can only be of length 1, or of the length of the maximum input. One way to do this may be: 
```
"Kf" <-
function(S=35,T=25,P=0,kf='x',pHscale="T",Ks_p0=0,Ks_p=0,warn="y"){
    # create vector of input lengths
    input_lengths = c(length(S), length(T), length(P), length(kf), length(pHscale), length(Ks_p0), length(Ks_p))

    # create nK, or max vector length
    nK = max(input_lengths)

    # stop function if any input length is not 1 or the max value in input_lengths. 
    # this error text is copied from CO2SYS. 
    if (sum(! input_lengths %in% c(1, nK))>0){
    stop('INPUT ERROR: Input vectors must all be of same length, or of length 1. ')    
    }
    
    if(length(S)!=nK){S <- rep(S[1], nK)}
    if(length(T)!=nK){T <- rep(T[1], nK)}
    if(length(P)!=nK){P <- rep(P[1], nK)}
    if(length(kf)!=nK){kf <- rep(kf[1], nK)}
    if(length(pHscale)!=nK){pHscale <- rep(pHscale[1], nK)}
 
    ### continue function as normal. 
```

This should only allow the rest of the function to run if the correct assumption is met: that all inputs have the same length, unless they are of length one. 

If this is of interest, I would be happy to fork and work on it. I thought I would ask before putting the work in.

**Comment by jpgattuso** [comment]
Sorry for my long silence. We are trying to fix an issue raised by CRAN. I will get back to you when that is done.

**Comment by jpgattuso** [comment]
I forgot: could you register as a participant?

**Comment by umihoshijima** [comment]
No problem, @jpgattuso - everything in this issue is relatively low-priority. I'm not sure exactly what you mean by "register as a participant"?

**Comment by jpgattuso** [comment]
Hi @umihoshijima sure, go ahead. I suggest that you do that using branch Candidate_3.2.5 in which I have tried to fix issue #54 raised by CRAN. Many thanks for spotting this problem and for proposing to fix it.

**Comment by jpgattuso** [comment]
@umihoshijima: can I close this issue?
